build/assets/scripts/pre-boot-tuning.sh

```diff
@@ -0,0 +1,123 @@
+
+# TODO - find a more robust approach than keeping the last timestamp
+RHCOS_OSTREE_PATH=$(ls -td /boot/ostree/*/ | head -1)
+RHCOS_OSTREE_BOOTLOADER_PATH=${RHCOS_OSTREE_PATH#"/boot"}
+INITRD_GENERATION_DIR="/root/initrd"
+INITRD_NEW_IMAGE="${RHCOS_OSTREE_PATH}/iso_initrd.img"
+
+# TODO: improve check for applied configuration
+if [ -f ${INITRD_NEW_IMAGE} ] && grep -qsR "iso_initrd.img" "/boot/loader/entries/"; then
+    echo "Pre boot tuning configuration already applied"
+    echo "Setting kernel rcuo* threads to the housekeeping cpus"
+    get_cpu_mask 1
+    pgrep rcuo* | while read line; do taskset -p $non_iso_cpumask $line || true; done
+else
+    # Clean up
+    rm -rf ${INITRD_GENERATION_DIR}
+    
+    # Create initrd image
+    mkdir ${INITRD_GENERATION_DIR}
+    mkdir -p ${INITRD_GENERATION_DIR}/usr/lib/dracut/hooks/pre-udev/
+    mkdir -p ${INITRD_GENERATION_DIR}/etc/systemd/
+    mkdir -p ${INITRD_GENERATION_DIR}/etc/sysconfig/
+    touch ${INITRD_GENERATION_DIR}/etc/systemd/system.conf
+    touch ${INITRD_GENERATION_DIR}/etc/sysconfig/irqbalance
+    touch ${INITRD_GENERATION_DIR}/usr/lib/dracut/hooks/pre-udev/00-tuned-pre-udev.sh
+    chmod +x ${INITRD_GENERATION_DIR}/usr/lib/dracut/hooks/pre-udev/00-tuned-pre-udev.sh
+
+    get_cpu_mask 1
+    echo '#!/bin/sh
+
+    type getargs >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+    #cpumask="$(getargs non_iso_cpumask)"
+    cpumask='$non_iso_cpumask'
+
+    log()
+    {
+    echo "tuned: $@" >> /dev/kmsg
+    }
+
+    if [ -n "$cpumask" ]; then
+    for file in /sys/devices/virtual/workqueue/cpumask /sys/bus/workqueue/devices/writeback/cpumask; do
+        log "setting $file CPU mask to $cpumask"
+        if ! echo $cpumask > $file 2>/dev/null; then
+        log "ERROR: could not write CPU mask for $file"
+        fi
+    done
+    fi' > ${INITRD_GENERATION_DIR}/usr/lib/dracut/hooks/pre-udev/00-tuned-pre-udev.sh
+
+    # Set CPU affinity according to NON_ISOLATED_CPUS
+    get_cpu_affinity
+    echo "[Manager]" >> ${INITRD_GENERATION_DIR}/etc/systemd/system.conf
+    echo "CPUAffinity=$cpu_affinity" >> ${INITRD_GENERATION_DIR}/etc/systemd/system.conf
+
+    # Set IRQ banned cpu according to NON_ISOLATED_CPUS
+    get_cpu_mask 0
+    echo "IRQBALANCE_BANNED_CPUS=$non_iso_cpumask" >> ${INITRD_GENERATION_DIR}/etc/sysconfig/irqbalance
+    
+    find ${INITRD_GENERATION_DIR} | cpio -co >${INITRD_NEW_IMAGE}
+
+    # Get current ostree config file according to the latest version
+    current_ver=1
+    entry_file=$(ls -td /boot/loader/entries/* | head -1)
+    while read -r line ; do
+        ver=`awk '/version/ {print $2}' $line`
+        if [ "$ver" -gt "$current_ver" ]; then
+           current_ver=$ver
+           entry_file=$line
+        fi
+    done <<<$(egrep $(uname -r) -lr /boot/loader/entries/)
+
+    sed -i "s^initrd .*\$^& ${RHCOS_OSTREE_BOOTLOADER_PATH}iso_initrd.img^" $entry_file
+
+    #TODO - once RHCOS image contains the initrd content we can set parameters with rpm-ostree:
```

Comment:
We might actually be able to test that already with latest RHCOS changes , but ill do it in another PR